tests/integration/apigateway/test_apigateway_api.py

```diff
@@ -69,6 +77,145 @@ def _factory(*args, **kwargs):
+            aws_client.apigateway,
+            restApiId=api_id,
+            resourceId=resource_id,
+            httpMethod="GET",
+            statusCode="200",
+        )
+        # Create the response for method POST /pets/{petId}
+        create_rest_api_method_response(
+            aws_client.apigateway,
+            restApiId=api_id,
+            resourceId=resource_id,
+            httpMethod="POST",
+            statusCode="200",
+        )
+        # Create the integration to connect GET /pets/{petId} to a backend
+        integration_uri, _ = create_rest_api_integration(
+            aws_client.apigateway,
+            restApiId=api_id,
+            resourceId=resource_id,
+            httpMethod="GET",
+            type="MOCK",
+            integrationHttpMethod="GET",
+            requestParameters={
+                "integration.request.path.id": "method.request.path.petId",
+            },
+            requestTemplates={"application/json": json.dumps({"statusCode": 200})},
+        )
+        # Create the integration to connect POST /pets/{petId} to a backend
+        integration_uri, _ = create_rest_api_integration(
+            aws_client.apigateway,
+            restApiId=api_id,
+            resourceId=resource_id,
+            httpMethod="POST",
+            type="MOCK",
+            integrationHttpMethod="POST",
+            requestParameters={
+                "integration.request.path.id": "method.request.path.petId",
+            },
+            requestTemplates={"application/json": json.dumps({"statusCode": 200})},
+        )
+        # Create the 200 integration response for GET /pets/{petId}
+        create_rest_api_integration_response(
+            aws_client.apigateway,
+            restApiId=api_id,
+            resourceId=resource_id,
+            httpMethod="GET",
+            statusCode="200",
+            responseTemplates={"application/json": json.dumps({"petId": "$input.params('petId')"})},
+        )
+        # Create the 200 integration response for POST /pets/{petId}
+        create_rest_api_integration_response(
+            aws_client.apigateway,
+            restApiId=api_id,
+            resourceId=resource_id,
+            httpMethod="POST",
+            statusCode="200",
+            responseTemplates={"application/json": json.dumps({"petId": "$input.params('petId')"})},
+        )
+
+        def invoke_method(path_with_query_string, method, body=""):
+            res = aws_client.apigateway.test_invoke_method(
+                restApiId=api_id,
+                resourceId=resource_id,
+                httpMethod=method,
+                pathWithQueryString=path_with_query_string,
+                body=body,
+            )
+            assert 200 == res.get("status")
+            return res
+
+        response = retry(
+            invoke_method, retries=10, sleep=5, path_with_query_string="/pets/123", method="GET"
+        )
+        snapshot.match("test-invoke-method-get", response)
```

Comment:
Do we maybe want to add 1-2 assertions for the content of `log` in the response? (as we're excluding the log details from the snapshot, which makes sense)

Reply:
Yep will do!